experimental/dds/tree2/src/domains/schemaBuilder.ts

```diff
@@ -343,6 +370,16 @@ export class SchemaBuilder<
 	 * {@link leaf.null}
 	 */
 	public readonly null = leaf.null;
+
+	/**
+	 * Function which can be used for its compile time sid-effects to tweak the evaluation order of recursive types to make them compile.
+	 * @remarks
+	 * Some related information in https://github.com/microsoft/TypeScript/issues/55758.
+	 *
+	 * Also be aware that code which relies on this (or the "recursive" SchemaBuilder methods tends to break VSCode's IntelliSense every time anything related to that code (even comments) is edited.
+	 * The command `TypeScript: Restart TS Server` should fix it.
+	 */
+	public fixRecursiveReference<T extends AllowedTypes>(...types: T): void {}
```

Comment:
Can we make this a free function instead? It seems odd as a method. I know it's not going to appear in Nick's demo if he doesn't use it, but still. Given the current doc, I am skeptical that anyone is going to discover this function organically and know what to do with it :D. So, it may as well be a free function that you have to know about to import.

Also, there should probably be a TODO of some kind here. E.g. "remove after the typescript problem is fixed, or improve docs and examples of this function if typescript cannot be fixed".

Reply:
If we want this function to have a sufficiently descriptive name to make sense at the top level, it will need an even longer name.

Also the only use case for this function when when you already have a builder.

And this whole builder API is optimized around reducing the number of things you need to import (ex: field kinds, leaf domain, TreeFieldSchema, etc). Adding a big long import seems like its violating the design principals we followed to get the import set of most schema files down to a single item.

Reply:
In practice no one will get either approach to recursive schema to work without cargoculting an example. Once we have decided which approach we want to endorse and are polishing up the docs, we will want to include full source examples in the doc comments here, but I think thats somthing to do later once we are more confident its not just gonning to become out of date or deleted.

Reply:
for now having this here lets us compare the two recursive schema approach and value how they would look to users (since if we use this, I really think we will want it here for import reduction, which I think is our current priority.

Reply:
But - we could do all of those things with a free function. Our priority is for import reduction _of things we want customers to use_. This is not one of those things, definitely not now, and hopefully not ever. It's _super weird_ and _not normal_; let's hide it! The rest of this PR is good so let's get it in, and if it's important to you that this goes in with it, then I'll be sad but fine we can remove it later :)